### ISSUE TITLE
Future visualization plans

### ISSUE THREAD

**Issue by ErikBjare** [ActivityWatch/aw-webui#17]
## General

We could definitely up our D3.js game. The [API reference](https://github.com/d3/d3/blob/master/API.md) would be a good place to start getting acquainted. 

Some highlights I found in the API documentation:

 - [Histograms](https://github.com/d3/d3-array/blob/master/README.md#histograms), useful when plotting application usage per day history. 
 - [Treemap](https://github.com/d3/d3-hierarchy/blob/master/README.md#treemap), not very visually pleasing but might be pretty practical. 
 - [Arcs](https://github.com/d3/d3-shape/blob/master/README.md#arcs), RescueTime uses something like this as their first visualization on their summary page.
   - Multilevel arcs/sunburst would also be cool, especially if you could get 2nd-level data like browser activity in. See also #25.
   - Really cool example: https://bl.ocks.org/kerryrodden/766f8f6d31f645c39f488a0befa1e3c8
 - [Sequential scales](https://github.com/d3/d3-scale/blob/master/README.md#sequential-scales), useful when coloring. 

## Improving the timeline

 - Implement zoom and pan (https://coderwall.com/p/psogia/simplest-way-to-add-zoom-pan-on-d3-js)
 - Show AFK time as grey to visualize inactivity
    - Would give a better impression of when there was activity, no need to hover and check the timestamp
 - Add time axis
 - How to display hierarchy in events?
   - Examples of hierarchies:
     - window -> title
     - browser window -> website -> url/title
   - Perhaps "fold out"/"slide down" a secondary detail timeline below the primary event when zoom level is high enough. 
   - The easiest thing to do is to just show everything in parallel timelines (probably do some filtering) and let the user sort out the hierarchy in their head from the overlap.
   - D3.js has tools for this: https://github.com/d3/d3-hierarchy/blob/master/README.md

## Calendar-view

~~See #16.~~

I found the following awesome calendar view (similar to the GitHub activity calendar thingy): https://bl.ocks.org/mbostock/4063318 (unfortunately GPLv3).

Could prove pretty useful, especially for date-selection.

**Comment by ErikBjare** [comment]
It should also be added that Google has a "My Activity" feature here: https://myactivity.google.com/

**Comment by johan-bjareholt** [comment]
@ErikBjare Wow, never seen that "myactivity" feature before. Incredibly creepy, even has saved all the recordings when i've searched something with google now on my phone.

**Comment by ErikBjare** [comment]
Found a cool timeline-like thing here: https://github.com/marmelab/EventDrops

**Comment by ErikBjare** [comment]
I started working on another version of the timeline. Currently used to visualize recent events in buckets. It works alright and I made a Vue.js component out of it, which is a nice way to deal with the packaging of these things.

See PR here: https://github.com/ActivityWatch/aw-webui/pull/35

**Comment by ErikBjare** [comment]
Also see the new separate issue on sunbursts: https://github.com/ActivityWatch/aw-webui/issues/36

**Comment by ErikBjare** [comment]
This was a really nice site: http://browser.timeviz.net/

**Comment by ErikBjare** [comment]
Someone already put down a lot of work into a basic GANTT-like timeline: https://www.npmjs.com/package/d3-timelines

Would be nice to try it out, all we'd had to add on top would be some filtering for AFK and webbrowser events.

**Comment by ErikBjare** [comment]
Here's a different multilane timeline that's zoomable: http://bl.ocks.org/TBD/600b23e56545026ae6fda2905efa42ce

**Comment by ErikBjare** [comment]
Almost all the things talked about in this issue has since been added or improved. Closing.
